### PR TITLE
+pro add compatibility test module

### PR DESCRIPTION
**Pull request by jrudolph**
It allows to write code in src/main against the previous version and run tests
in src/test with the new source code version on the classpath.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by jrudolph** [comment]
(Autoformatter currently cannot be CLA verified)

File: build.sbt [hunk 1]
```diff
@@ -360,4 +361,17 @@ lazy val docs = project("docs")
   )
   .settings(ParadoxSupport.paradoxWithCustomDirectives)
 
+lazy val compatibilityTests = Project("akka-http-compatibility-tests", file("akka-http-compatibility-tests"))
+  .enablePlugins(NoPublish)
+  .disablePlugins(BintrayPlugin, MimaPlugin)
+  .settings(
+    libraryDependencies += "com.typesafe.akka" %% "akka-http" % "10.1.8" % "provided", // TODO, should we make that latest?
```
Comment (raboof) [inline build.sbt]: Probably?

File: build.sbt [hunk 2]
```diff
@@ -360,4 +361,17 @@ lazy val docs = project("docs")
...
+    (dependencyClasspath in Test) := {
+      // HACK: We'd like to use `dependsOn(http % "test->compile")` to upgrade the explicit dependency above to the
+      //       current version but that fails. So, this is a manual `dependsOn` which works as expected.
```
Comment (raboof) [inline build.sbt]: Hm, weird

File: build.sbt (hunk 1 above)
Comment (jrudolph) [reply build.sbt]: Maybe let's keep it at 10.1.8 for a while which was a version that was alive for a long time.

**Pull request merged**